src/wifi.rs

```diff
@@ -1050,6 +1086,42 @@ impl<'d> WifiDriver<'d> {
         esp!(unsafe { esp_wifi_set_mac(interface.into(), mac.as_ptr() as *mut _) })
     }
 
+    /// Enable and start WPS
+    pub fn start_wps(&mut self, config: &WpsConfig) -> Result<(), EspError> {
```

Comment:
`start_wps` / `take_wps_event` is confusing. Why not:
- `start_wps` - calls `esp_wifi_wps_enable` and then `esp_wifi_wps_start` (and `None`-s the `WpsStatus instance kept in the driver)
- `stop_wps` - calls `esp_wifi_wps_disable`
- `get_wps_status(...) -> WpsStatus` (WpsStatus is your renamed WpsEvent) - returns the current WPS status (including the received WPS credentials, if any)

Reply:
Optionally, we can also merge `get_wps_status` with `stop_wps`